python/ray/workflow/tests/test_metadata.py

```diff
@@ -296,11 +296,8 @@ def get(self):
             return self.n
 
     counter = Counter.get_or_create("counter")
-    counter.incr.options(name="incr", metadata={"outer_k": "outer_v"}).run(5)
```

Comment:
why do we remove this?

Reply:
Ok. it seems be moved to `test_basic_workflows.py`

Reply:
@iycheng with the new .options() support user_metadata "update", adding this will make recursive results changing from:
```
assert workflow.get_metadata("counter", "incr_3")["user_metadata"] == {"current_n": 3}
```
to
```
assert workflow.get_metadata("counter", "incr_3")["user_metadata"] == {"outer_k": "outer_v", "current_n": 3}
```
so either remove the outer metadata or update the assert result.


In fact, one thing that is missing is the test on new options behavior for virtual actor.  maybe I should add it.